addons/account_edi/models/account_edi_document.py

```diff
@@ -186,7 +186,7 @@ def _postprocess_cancel_edi_results(documents, edi_result):
                 move = document.move_id
                 move_result = edi_result.get(move, {})
                 if move_result.get('success') is True:
-                    old_attachment = document.attachment_id
+                    old_attachment = document.sudo().attachment_id
```

Comment:
do you think we might need the same on lines 161 and 260?

Reply:
It seems not necessary, as in these cases there is no attachment or the attachment has `res_id` and the action will not be blocked